llvm/lib/Target/LoongArch/Disassembler/LoongArchDisassembler.cpp

```diff
@@ -157,6 +157,29 @@ static DecodeStatus decodeSImmOperand(MCInst &Inst, uint64_t Imm,
   return MCDisassembler::Success;
 }
 
+// Decode AMOSWAP.W and UD, which share the same base encoding.
+// If rk == 1 and rd == rj, interpret the instruction as UD;
+// otherwise decode as AMSWAP.W.
+static DecodeStatus DecodeAMOrUDInstruction(MCInst &Inst, unsigned Insn,
+                                            uint64_t Address,
+                                            const MCDisassembler *Decoder) {
+  unsigned rd = fieldFromInstruction(Insn, 0, 5);
+  unsigned rj = fieldFromInstruction(Insn, 5, 5);
+  unsigned rk = fieldFromInstruction(Insn, 10, 5);
```

Comment:
Consistent with existing code.
```suggestion
  unsigned Rd = fieldFromInstruction(Insn, 0, 5);
  unsigned Rj = fieldFromInstruction(Insn, 5, 5);
  unsigned Rk = fieldFromInstruction(Insn, 10, 5);
```